core/modules/user/store/actions.ts

```diff
@@ -1,379 +1,299 @@
     if (config.usePriceTiers) {
       if (userData.groupToken) {
-        context.commit(types.USER_GROUP_TOKEN_CHANGED, userData.groupToken)
+        commit(types.USER_GROUP_TOKEN_CHANGED, userData.groupToken)
       }
 
       if (userData.group_id) {
-        context.commit(types.USER_GROUP_CHANGED, userData.group_id)
+        commit(types.USER_GROUP_CHANGED, userData.group_id)
       }
     } else {
-      context.commit(types.USER_GROUP_TOKEN_CHANGED, '')
-      context.commit(types.USER_GROUP_CHANGED, null)
+      commit(types.USER_GROUP_TOKEN_CHANGED, '')
+      commit(types.USER_GROUP_CHANGED, null)
+    }
+  },
+  async restoreCurrentUserFromCache ({ commit, dispatch }) {
+    const usersCollection = StorageManager.get('user')
+    const currentUser = await usersCollection.getItem('current-user')
+
+    if (currentUser) {
+      commit(types.USER_INFO_LOADED, currentUser)
+      dispatch('setUserGroup', currentUser)
+      EventBus.$emit('user-after-loggedin', currentUser)
+      dispatch('cart/authorize', {}, { root: true })
+
+      return currentUser
+    }
+
+    return null
+  },
+  async refreshUserProfile ({ commit, dispatch }, { resolvedFromCache }) {
+    const resp = await UserService.getProfile()
+
+    if (resp.resultCode === 200) {
+      commit(types.USER_INFO_LOADED, resp.result) // this also stores the current user to localForage
+      dispatch('setUserGroup', resp.result)
+    }
+
+    if (!resolvedFromCache && resp.resultCode === 200) {
+      EventBus.$emit('user-after-loggedin', resp.result)
+      dispatch('cart/authorize', {}, { root: true })
+      Promise.resolve(resp)
+    } else {
+      Promise.resolve(null)
     }
+
+    return resp
   },
   /**
    * Load current user profile
    */
-  me (context, { refresh = true, useCache = true } = {}) {
-    return new Promise((resolve, reject) => {
-      if (!context.state.token) {
-        Logger.warn('No User token, user unauthorized', 'user')()
-        return resolve(null)
-      }
-      const cache = StorageManager.get('user')
-      let resolvedFromCache = false
-
-      if (useCache === true) { // after login for example we shouldn't use cache to be sure we're loading currently logged in user
-        cache.getItem('current-user', (err, res) => {
-          if (err) {
-            Logger.error(err, 'user')()
-            return
-          }
-
-          if (res) {
-            context.commit(types.USER_INFO_LOADED, res)
-            context.dispatch('setUserGroup', res)
-            EventBus.$emit('user-after-loggedin', res)
-            rootStore.dispatch('cart/authorize')
-
-            resolve(res)
-            resolvedFromCache = true
-            Logger.log('Current user served from cache', 'user')()
-          }
-        })
+  async me ({ dispatch, getters }, { refresh = true, useCache = true } = {}) {
+    if (!getters.getToken) {
+      Logger.warn('No User token, user unauthorized', 'user')()
+      return null
```

Comment:
it's default that promise returns undefined
```suggestion
      return
```